monai/bundle/scripts.py

```diff
@@ -116,6 +121,216 @@ def _get_fake_spatial_shape(shape: Sequence[Union[str, int]], p: int = 1, n: int
+        if filepath.name.endswith(("zip", "tar", "tar.gz")):
+            extractall(filepath=filepath, output_dir=bundle_dir_, has_base=True)
+    elif source_ == "github":
+        _download_from_github(
+            repo=repo_, tag_name=name_, download_path=bundle_dir_, filename=filename, progress=progress_
+        )
+    else:
+        raise NotImplementedError("So far, only support to download from url or `github` source.")
+
+
+def load(
+    name: Optional[str] = None,
+    is_ts_model: bool = False,
+    bundle_dir: Optional[PathLike] = None,
+    source: str = "github",
+    repo: Optional[str] = None,
+    url: Optional[str] = None,
+    progress: bool = True,
+    map_location=None,
+    net_name: Optional[str] = None,
+    **net_kwargs,
+):
+    """
+    Load model weights. If the weights file is not existing locally, it will be downloaded first. The function can
+    return the weights, or an instantiated network that loaded the weights.
+
+    Args:
+        name: Bundle and weights name. If `None`, `url` should be provided, or the weights file is existing locally and
+            the default weights file should be named as "model.pt".
+            If not `None`, it should be a string that consisted with the bundle name, a separator `#`
+            and the weights name. For example: if the bundle name is "spleen" and the weights name is "model.pt", then
+            `name` is "spleen#model.pt".
+        is_ts_model, a flag to specify if the weights file is a TorchScript module.
+        bundle_dir: the directory the weights will be loaded from.
+            Default is `bundle` subfolder under`torch.hub get_dir()`.
+        source: the place that saved the bundle.
+            If `source` is `github`, the bundle should be within the releases.
+        repo: the repo name. If the weights need to be downloaded first and `url` is `None`, it must be provided.
+        url: the url to download the data.
+        progress: whether to display a progress bar when downloading.
+        map_location: pytorch API parameter for `torch.load` or `torch.jit.load`.
+        net_name: if not `None`, a corresponding network will be instantiated and load the achieved weights.
+            This argument only works when loading weights.
+        net_kwargs: other arguments that are used to instantiate the network class defined by `net_name`.
+
+    """
+    if bundle_dir is None:
+        get_dir, has_home = optional_import("torch.hub", name="get_dir")
+        if has_home:
+            bundle_dir = Path(get_dir()) / "bundle"
+        else:
+            raise ValueError("bundle_dir=None, but no suitable default directory computed. Upgrade Pytorch to 1.6+ ?")
+    bundle_dir = Path(bundle_dir)
+
+    weights_name: str = "model.pt"
+    if name is not None:
+        if len(name.split(ID_SEP_KEY)) == 2:
+            weights_name = name.split(ID_SEP_KEY)[1]
+        else:
+            raise ValueError(f"The format of `name` is wrong.\n{run.__doc__}")
```

Comment:
This logic is not correct and your unit tests don't cover it.
If the `name` is the whole bundle name without `#`, `len(name.split(ID_SEP_KEY))` will be 1.

Thanks.